ONSdigital/ssdc-rm-support-tool
# SDCSRM-289 Support Tool Dummy User Cleanup

# Motivation and Context
Dummy User in Support Tool needs cleaning up.
We wanted a way to have the Dummy User code not be visible in prod environments and for it to be 'injected' only in dev and test environments

# What has changed
 - Made two new classes - one that handles/validates IAP Users and one that handles/validated DummyUsers that both implement a new interface `AuthUser`. 
- replaced env var `DUMMYUSERIDENTITY-ALLOWED` with `IAP_ENABLED` which is used to see if a dummyUser is allowed
- Using 'Spring magic'; depending on the env only one of the classes will be loaded in UserIdentity and AuthorisationEndpoint
- Removed all Dummy User code that isn't in the new DummyUser Class

# How to test?
Run `make build` to see it passes tests using a dummy user
There's a [docker-dev PR](https://github.com/ONSdigital/ssdc-rm-docker-dev/pull/101) that has the new env needed for local dev testing
There's also a [Kubernetes PR](https://github.com/ONSdigital/ssdc-rm-kubernetes/pull/194) for gcp environments
Run the ATs against it (it should use IAP)

# Links
[SDCSRM-289](https://jira.ons.gov.uk/browse/SDCSRM-289)
